primitives/arithmetic/src/per_things.rs

```diff
@@ -64,12 +112,43 @@ pub trait PerThing: Sized + Saturating + Copy {
 	/// `M` as the data type that can hold the maximum value of this per-thing (e.g. u32 for
 	/// perbill), this can only work if `N == M` or `N: From<M> + TryInto<M>`.
 	fn from_rational_approximation<N>(p: N, q: N) -> Self
-		where N: Clone + Ord + From<Self::Inner> + TryInto<Self::Inner> + ops::Div<N, Output=N>;
+	where N: Clone + Ord + From<Self::Inner> + TryInto<Self::Inner> + ops::Div<N, Output=N>;
+}
+
+/// Helper function to calculate a remainder used in `PerThing` multiplications.
+fn rem_helper<N, Inner, Upper>(x: N, part: Inner, acc: Inner, reciprocal: bool, truncate: bool) -> Upper 
```

Comment:
could use a better name that tells what this function does.

Reply:
Also, once you name it better it because clear what this function is doing, I imagine the parameters could also use a re-think. I don't think it is the responsibility of this function (if it is merely `rem`ing two numbers) to know about `reciprocal: bool`.

Reply:
Good point. the caller could just swap `part` and `acc` to get the same result